src/main/java/edu/kis/powp/jobs2d/drivers/adapter/LineDriverAdapter.java

```diff
@@ -31,6 +33,9 @@ public void operateTo(int x, int y) {
         line.setStartCoordinates(this.startX, this.startY);
         this.setPosition(x, y);
         line.setEndCoordinates(x, y);
+        try {
+            linesRecorder.addLine((ILine) line.clone());
+        } catch (CloneNotSupportedException ignored) {}
```

Comment:
The solution would be better if LinesRecorder dependency could be avoided. Maybe, the decorator approach is possible by providing a 'line' clone getter in LineDriverAdapter.